idp-service/src/test/java/io/mosip/idp/flows/AuthorizationAPIFlowTest.java

```diff
@@ -0,0 +1,432 @@
+        Assert.assertNotNull(authResponseResponseWrapper);
+        Assert.assertNotNull(authResponseResponseWrapper.getResponseTime());
+        Assert.assertNotNull(authResponseResponseWrapper.getResponse());
+        Assert.assertEquals(oAuthDetailResponse.getTransactionId(), authResponseResponseWrapper.getResponse().getTransactionId());
+
+        ResponseWrapper<AuthCodeResponse> responseWrapper = getAuthCode(oAuthDetailResponse.getTransactionId(), state, nonce);
+        Assert.assertNotNull(responseWrapper);
+        Assert.assertNotNull(responseWrapper.getResponseTime());
+        Assert.assertNotNull(responseWrapper.getResponse());
+        Assert.assertNotNull(responseWrapper.getResponse().getCode());
+        Assert.assertEquals(state, responseWrapper.getResponse().getState());
+        Assert.assertEquals(nonce, responseWrapper.getResponse().getNonce());
+        Assert.assertEquals(redirectionUrl, responseWrapper.getResponse().getRedirectUri());
+
+        ResponseWrapper<OtpResponse> otpResponseResponseWrapper = sendOtp(authResponseResponseWrapper.getResponse().getTransactionId(),
+                "8267411571");
+        assertErrorCode(otpResponseResponseWrapper, ErrorConstants.INVALID_TRANSACTION);
+    }
+
+    @Test
+    public void authorizationCodeFlowTest_sendotp_after_failed_authcode_call() throws Exception {
+        ResponseWrapper<OAuthDetailResponse> oAuthDetailResponseWrapper = getOauthDetails(clientId, redirectionUrl, state, nonce);
+        OAuthDetailResponse oAuthDetailResponse = oAuthDetailResponseWrapper.getResponse();
+
+        ResponseWrapper<AuthResponse> authResponseResponseWrapper = authenticate(oAuthDetailResponse.getTransactionId());
+        Assert.assertNotNull(authResponseResponseWrapper);
+        Assert.assertNotNull(authResponseResponseWrapper.getResponseTime());
+        Assert.assertNotNull(authResponseResponseWrapper.getResponse());
+        Assert.assertEquals(oAuthDetailResponse.getTransactionId(), authResponseResponseWrapper.getResponse().getTransactionId());
+
+        ResponseWrapper<AuthCodeResponse> responseWrapper = getAuthCode("weerwerwerwe", state, nonce);
+        assertErrorCode(responseWrapper, ErrorConstants.INVALID_TRANSACTION);
+
+        ResponseWrapper<OtpResponse> otpResponseResponseWrapper = sendOtp(authResponseResponseWrapper.getResponse().getTransactionId(),
+                "8267411571");
+        assertErrorCode(otpResponseResponseWrapper, ErrorConstants.INVALID_TRANSACTION);
+    }
+
+    @Test
+    public void authorizationCodeFlowTest_sendotp_after_failed_auth_call() throws Exception {
+        ResponseWrapper<OAuthDetailResponse> oAuthDetailResponseWrapper = getOauthDetails(clientId, redirectionUrl, state, nonce);
+        OAuthDetailResponse oAuthDetailResponse = oAuthDetailResponseWrapper.getResponse();
+
+        ResponseWrapper<AuthResponse> authResponseResponseWrapper = authenticate("failed-auth");
```

Comment:
Should we not send the transaction id here ?

Reply:
yes. corrected.